src/test/java/org/takes/http/BkParallelTest.java

```diff
@@ -23,15 +23,33 @@
  */
 package org.takes.http;
 
+import java.net.Socket;
+import java.util.concurrent.ExecutorService;
+import org.junit.Test;
+import org.mockito.Mockito;
+
 /**
  * Test case for {@link BkParallel}.
  *
  * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
  * @version $Id$
  * @since 0.15.2
- * @todo #220:30min/DEV This unit test not implemented but has to be.
- *  E.g. you could pass your own a ExecutorService implementation to the ctor
- *  and verify execute() calls.  But feel free to try another way.
  */
-public class BkParallelTest {
+public final class BkParallelTest {
+    /**
+     * BkParallel can call execute() when called accept().
+     */
+    @Test
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    public void callsExecuteOnAccept() {
```

Comment:
this is a very bad design of a unit test. the puzzle is wrong, we should NOT mock the executor service. in general, mocking is a bad practice: http://www.yegor256.com/2014/09/23/built-in-fake-objects.html Instead, we should create a full integration test here, which will make a few requests in parallel threads and make sure that all of them are resolved in parallel. how it's done inside `BkParallel` should not be on any concern to the test